velox/connectors/hive/storage_adapters/abfs/AbfsWriteFile.cpp

```diff
@@ -120,16 +120,24 @@ class AbfsWriteFile::Impl {
 AbfsWriteFile::AbfsWriteFile(
     std::string_view path,
     const config::ConfigBase& config) {
-  auto abfsAccount = AbfsConfig(path, config);
-  std::unique_ptr<AzureDataLakeFileClient> client =
-      std::make_unique<DataLakeFileClientWrapper>(
-          std::make_unique<DataLakeFileClient>(
-              DataLakeFileClient::CreateFromConnectionString(
-                  abfsAccount.connectionString(),
-                  abfsAccount.fileSystem(),
-                  abfsAccount.filePath())));
-
-  impl_ = std::make_unique<Impl>(path, client);
+  auto abfsConfig = AbfsConfig(path, config, true);
+  std::unique_ptr<AzureDataLakeFileClient> clientWrapper;
+  if (abfsConfig.authType() == "SAS") {
```

Comment:
Let's add `std::unique_ptr<AzureDataLakeFileClient> getWriteFileClient()` to AbfsConfig.

Reply:
Updated.